Packs/OpenAI/Integrations/OpenAI/OpenAI.yml

```diff
@@ -3,62 +3,115 @@ commonfields:
+category: Messaging and Conferencing
+description: This integration leverages the OpenAI and Azure OpenAI REST APIs to generate human-like responses to text prompts. The APIs provide access to OpenAI's powerful language models including the GPT-3.5 model series. These models can be easily adapted to your specific task including but not limited to content generation, summarization, semantic search, and natural language to code translation.
 configuration:
-- display: OpenAI API URL(e.g. https://api.openai.com/)
+- display: 'Server URL: (e.g. https://api.openai.com/)'
   name: url
   defaultvalue: https://api.openai.com/
   type: 0
   required: true
+  section: Connect
 - display: API Key
   name: apikey
   type: 4
   required: true
+  additionalinfo: OpenAI API key
+- display: Use Azure OpenAI?
+  name: is_azure
+  type: 8
+  required: false
+  additionalinfo: Whether to use Azure OpenAI (https://your-resource-name.openai.azure.com) instead of standard OpenAI (https://api.openai.com)
+  defaultvalue: "false"
 - display: Trust any certificate (not secure)
   name: insecure
   type: 8
   required: false
 - display: Use system proxy settings
   name: proxy
+  required: false
   type: 8
+- additionalinfo: Used for Azure OpenAI only. Default value is "2023-09-01-preview".
+  defaultvalue: 2023-09-01-preview
+  display: API Version
+  name: version
   required: false
+  type: 0
+- additionalinfo: Model to use for the integration Test if different from the OpenAI API default Chat model
+  display: Model to Use for Test
+  name: test_model
+  required: false
+  type: 0
 script:
   script: ''
   type: python
   commands:
-  - name: openai-completions
+  - name: openai-chatgpt
     arguments:
     - name: prompt
       required: true
-      description: Instruction
+      description: Add your question or text.
     - name: model
       auto: PREDEFINED
       predefined:
+      - gpt-4
+      - gpt-3.5-turbo
+      description: Name of model to use. For Azure OpenAI, this should be the name of the DEPLOYMENT, not the generic model name.
+      defaultValue: gpt-3.5-turbo
+    outputs:
+    - contextPath: OpenAI.ChatGPTResponse.ChatGPT Response
+      description: ChatGPT response.
+    - contextPath: OpenAI.ChatGPTResponse.Created Time
+      description: Response created time as Unix timestamp.
+    - contextPath: OpenAI.ChatGPTResponse.Model
+      description: Model that generated the response.
+    - contextPath: OpenAI.ChatGPTResponse.Number of Completion Tokens
+      description: Number of tokens in the completion (response).
+    - contextPath: OpenAI.ChatGPTResponse.Number of Prompt Tokens
+      description: Number of tokens in the prompt (input).
+    - contextPath: OpenAI.ChatGPTResponse.Number of Total Tokens
+      description: Total number of tokens in the completion and prompt.
+    - contextPath: OpenAI.ChatGPTResponse.id
```

Comment:
```suggestion
```suggestion
    - contextPath: OpenAI.ChatGPTResponse.ChatGPT_Response
      description: ChatGPT response.
    - contextPath: OpenAI.ChatGPTResponse.Created_Time
      description: Response created time as Unix timestamp.
    - contextPath: OpenAI.ChatGPTResponse.Model
      description: Model that generated the response.
    - contextPath: OpenAI.ChatGPTResponse.Number_of_Completion_Tokens
      description: Number of tokens in the completion (response).
    - contextPath: OpenAI.ChatGPTResponse.Number_of_Prompt_Tokens
      description: Number of tokens in the prompt (input).
    - contextPath: OpenAI.ChatGPTResponse.Number_of_Total_Tokens
      description: Total number of tokens in the completion and prompt.
    - contextPath: OpenAI.ChatGPTResponse.id
```
It is not recommended to use space in the context path